### PR TITLE
CI: Test pyarrow nightly instead of intermediate versions

### PR DESCRIPTION
**Pull request by mroeschke**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: ci/deps/actions-311-pyarrownightly.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  - versioneer[toml]
+  - cython>=0.29.33
+
+  # test dependencies
+  - pytest>=7.0.0
+  - pytest-cov
+  - pytest-xdist>=2.2.0
+  - hypothesis>=6.34.2
+  - pytest-asyncio>=0.17.0
+  - boto3
```
Comment (lithomas1) [inline ci/deps/actions-311-pyarrownightly.yaml:17]: Is boto3 a required dep for pyarrow?

Comment (mroeschke) [reply ci/deps/actions-311-pyarrownightly.yaml]: Don't think so. I think i accidentally copied this from another deps file

File: pandas/io/parquet.py [hunk 2]
```diff
@@ -92,22 +92,18 @@ def _get_path_or_handle(
     if fs is not None:
         pa_fs = import_optional_dependency("pyarrow.fs", errors="ignore")
         fsspec = import_optional_dependency("fsspec", errors="ignore")
-        if pa_fs is None and fsspec is None:
```
Comment (lithomas1) [inline pandas/io/parquet.py:95]: Why did this change

Comment (mroeschke) [reply pandas/io/parquet.py:95]: The build dependency setup uncovered a bug in the recent pyarrow Filesystem implementation not raising a `ValueError` consistently if pyarrow was install but fastparquet wasn't

**Review by lithomas1 — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Do we have an issue to add back testing with multiple versions of pyarrow? I understand this PR reduced the CI load, but now we have several supported versions of pyarrow that we are not testing at all. 

And it actually seems that in practice, something else in the environment is causing a constraint on older pyarrow, so all our standard builds are using pyarrow 9.0.0. Which means we are not even testing the latest released pyarrow version at the moment?

We have standard env files for python 3.9, 3.10 and 3.11 at the moment, we could easily test three different pyarrow versions in those? (xref https://github.com/pandas-dev/pandas/pull/51308#issuecomment-1426406042)

**Comment by jorisvandenbossche** [comment]
Testing our actions-311.yaml env file locally, and then trying to upgrade pyarrow causes a downgrade of pandas-gbq, so the bump of the minimum version of this might have caused to test with an older version of pyarrow.

**Comment by mroeschke** [comment]
We don't have an open issue to discuss adding back testing with multiple version, but it would be good to have one to discuss. Yeah maybe we could have jobs that tests the required dependencies + pyarrow 8, 9, 10, ...

**Comment by lithomas1** [comment]
> Do we have an issue to add back testing with multiple versions of pyarrow? I understand this PR reduced the CI load, but now we have several supported versions of pyarrow that we are not testing at all.

Are there issues with pyarrow versions in the middle of our support range breaking?

FWIW, for numpy we only test the min, the latest, and numpy dev, and it seems to work fine.

Although the build matrix has been reduced by fusing jobs together, the individual jobs themselves take longer now (~1hr per job), so I'm still a little worried about the runners clogging up (We support a wide range of pyarrow rn, which is pyarrow 7-12 I think). 


> And it actually seems that in practice, something else in the environment is causing a constraint on older pyarrow, so all our standard builds are using pyarrow 9.0.0. Which means we are not even testing the latest released pyarrow version at the moment?

This seems like an issue. 

We could try adding a "clean" job without any dependencies, that would have pyarrow in it (this depends on PDEP-10 being accepted).

In this case though, we are probably just pulling a very old version of pandas-gbq (mamba bug?).

**Pull request merged**